app/controllers/dashboards_controller.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+# Controller for Dashboards
+class DashboardsController < ApplicationController
+  before_action :authorize!
+
+  def show
+    admin_metrics unless current_user.student?
+    student_variables unless current_user.admin?
+  end
+
+  private
+
+  def admin_metrics
+    @draws = Draw.all.includes(:groups).sort_by(&:name)
+    @draw_hash = Hash.new { |hash, key| hash[key] = {} }
+    @draws.each do |d|
+      @draw_hash[d][:partial] = d.status
+      if %w(pre_lottery suite_selection results).include? d.status
+        send("#{d.status}_metrics".to_sym, d, @draw_hash[d])
+      end
+    end
+  end
+
+  # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+  def pre_lottery_metrics(draw, hash)
+    suite_counts = draw.suites.available.group(:size).count
+    suite_counts.default = 0
+    groups = draw.groups.sort_by { |g| Group.statuses[g.status] }
+                 .group_by(&:size)
+    groups.default = []
+    group_counts = groups.map { |s, g| [s, g.count] }.to_h
+    group_counts.default = 0
+    sizes = (suite_counts.keys + group_counts.keys).uniq.sort
+    diff = sizes.map { |s| [s, suite_counts[s] - group_counts[s]] }.to_h
+
+    locked_counts = draw.groups.where(status: 'locked').group(:size).count
```

Comment:
this is the simplest way to do it because we have to actually drop values from the `groups` hash, which isn't particularly nice to do:
```ruby
    locked_counts = groups.each_with_object({}) do |(k, v), h|
      h[k] = v.select(&:locked?)
    end.transform_values(&:count)
```